pkg/transformer/openshift/openshift.go

```diff
@@ -202,11 +203,20 @@ func (o *OpenShift) initImageStream(name string, service kobject.ServiceConfig,
 func initBuildConfig(name string, service kobject.ServiceConfig, repo string, branch string) (*buildapi.BuildConfig, error) {
 	contextDir, err := getAbsBuildContext(service.Build)
 	envList := []kapi.EnvVar{}
-	for envName, envValue := range service.BuildArgs {
-		if *envValue == "\x00" {
-			*envValue = os.Getenv(envName)
+	var sortEnvName []string
+	for EnvName, _ := range service.BuildArgs {
+		sortEnvName = append(sortEnvName, EnvName)
+	}
+	sort.Strings(sortEnvName)
+	for _, newEnv := range sortEnvName {
+		for envName, envValue := range service.BuildArgs {
+			if envName == newEnv {
+				if *envValue == "\x00" {
+					*envValue = os.Getenv(envName)
+				}
+				envList = append(envList, kapi.EnvVar{Name: envName, Value: *envValue})
+			}
```

Comment:
Above function is way too complex (two nested loops, nested if statements, etc.), no comments as well as confusing as to what is happening. 

It'd be better to split this off into it's own functional as well as separate PR (ex: PR saying "fix ordering of environment values").